tests/ci/check_so_interface.py

```diff
@@ -0,0 +1,352 @@
+                ],
+                "defaults": [
+                    'None', '2 ** 12', "'constant'", 'None', 'None', 'None', 'True', 'True', 'False', 'False', 'False', 'None', 'None', 'True', 'True', 'True', 'False', "''"
+                ]
+            }
+        },
+    }
+}
+
+"""
+Frozen Functions:
+
+SETUP FUNCTIONS
+setup
+set_amplifier_bias #smurf_command
+set_cryo_card_ps_en #smurf_command
+which_on #smurf_util
+band_off #smurf_util
+channel_off #smurf_util
+
+TUNING FUNCTIONS
+full_band_resp #smurf_tune.py
+find_freq #smurf_tune.py
+setup_notches #smurf_tune.py
+run_serial_gradient_descent #smurf_command
+run_serial_eta_scan #smurf_command
+plot_tune_summary #smurf_tune.py
+tracking_setup #smurf_tune.py
+set_amplitude_scale_array #smurf_command
+
+TES/FLUX RAMP FUNCTIONS
+set_tes_bias_bipolar_array #smurf_util
+set_tes_bias_high_current #smurf_util
+set_tes_bias_low_current #smurf_util
+set_mode_dc #smurf_util
+set_mode_ac #smurf_util
+flux_ramp_setup #smurf_tune.py
+
+DATA ACQUISITION FUNCTIONS
+set_stream_enable #smurf_command
+take_stream_data #smurf_util
+take_noise_psd #smurf_noise
+stream_data_on #smurf_util
+stream_data_off #surf_util
+read_stream_data #smurf_util
+set_downsample_filter #smurf_util
+
+IV FUNCTIONS
+run_iv #smurf_iv
+analyze_iv #smurf_iv
+
+DATA OUTPUTS TO DISK
+tune files generated when new resonators are found
+channel mapping file format
+.dat noise files - generated by take_stream_data
+iv_files - generated by run_iv"
+"""
+def compare_args(node, fname, classname, funcname, type):
+    # Check function
+    if type == "isfunc":
+        spec_args = interface[fname][classname][funcname]["args"]
+        spec_defaults = interface[fname][classname][funcname]["defaults"]
+
+        # Extract function arguments from the AST node
+        found_args = [arg.arg for arg in node.args.args]
+        found_defaults = [ast.unparse(d) if d else None for d in node.args.defaults]
+
+        # Compare arguments to specified arguments in interface dict
+        if found_args != spec_args:
+            print(f"Mismatch in arguments for function {funcname}")
+            print(f"Found: {found_args}")
+            print(f"Expected: {spec_args}")
+            return False
+        
+        # Compare default values to specified defaults in interface dict
+        if found_defaults != spec_defaults:
+            print(f"Mismatch in default values for function {funcname}")
+            print(f"Found: {found_defaults}")
+            print(f"Expected: {spec_defaults}")
+            return False
+    # Check class
+    elif type == "isclass":
+        spec_meths = interface[fname][classname]["methods"]
+
+        # Extract class methods from the AST node
+        found_meths = []
+        for child in node.body:
+            if isinstance(child, ast.FunctionDef):
+                found_meths.append(child.name)
+
+        # Compare methods to specified methods in interface dict
+        if found_meths != spec_meths:
```

Comment:
In this case we don't care about the order of methods. Consider using the `set` built-in and check that the intersection of the two is complete.